store/tikv/gcworker/gc_worker.go

```diff
@@ -560,28 +735,30 @@ func doGC(ctx goctx.Context, store tikv.Storage, safePoint uint64, identifier st
 		}
 
 		if err != nil {
+			log.Warnf("[gc worker] gc for range [%v, %v) safepoint: %v, failed, err: %v", startKey, endKey, safePoint, err)
+			lastErr = err
 			failedRegions++
-			gcActionRegionResultCounter.WithLabelValues("fail").Inc()
-			log.Warnf("[gc worker] %s failed to do gc on region(%s, %s), ignore it", identifier, string(loc.StartKey), string(loc.EndKey))
 		} else {
 			successRegions++
-			gcActionRegionResultCounter.WithLabelValues("success").Inc()
 		}
 
 		key = loc.EndKey
```

Comment:
Is [startkey, endkey]  a range and should we consider all regions in the range?

Reply:
Yes, we do it in a range, not just a region if it splited